platform/android/display_server_android.cpp

```diff
@@ -243,6 +243,24 @@ DisplayServer::WindowID DisplayServerAndroid::get_window_at_screen_position(cons
 	return MAIN_WINDOW_ID;
 }
 
+int64_t DisplayServerAndroid::window_get_native_handle(HandleType p_handle_type, WindowID p_window) const {
+	ERR_FAIL_COND_V(p_window != MAIN_WINDOW_ID, 0);
+	switch (p_handle_type) {
+		case DISPLAY_HANDLE: {
+			return 0; // Not supported.
+		}
+		case WINDOW_HANDLE: {
+			return (int64_t)((OS_Android *)OS::get_singleton())->get_godot_java()->get_activity();
+		}
```

Comment:
I don't see a similar implementation in the 3.x branch. @BastiaanOlij Was there a plan to add something similar to 3.x?

Reply:
>			- MacOS: [code]NSWindow*[/code] of the main window (not yet implemented)
>			- Android: [code]jObject[/code] the main android activity (not yet implemented)

It's (and macOS code) in the 3.x docs, but seems like it was never implemented.

Reply:
I think for android we already had access to these through GDNative API extensions, it makes sense for consistency to move them into the `get_driver_resource()`.

macOS was never implemented because I have no way of testing it anymore, plus that there currently are no XR implementations working on macOS.